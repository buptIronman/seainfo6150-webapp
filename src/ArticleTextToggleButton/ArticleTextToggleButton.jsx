import React, {useState} from 'react';
import styles from './ArticleTextToggleButton.module.css';

const ArticleTextToggleButton = ({ButtonText, onClick}) => {

    const [buttonState, setButtonState] = useState(false);


    return (<div>
        <button onClick={() => {
            console.log(1)
            onClick(buttonState);
            setButtonState(!buttonState);
            console.log(buttonState);
        }} className={styles.button}>{ButtonText}</button>
    </div>)
}


export default ArticleTextToggleButton;