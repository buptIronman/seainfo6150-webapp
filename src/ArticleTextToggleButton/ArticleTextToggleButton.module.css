.button{
    border-style: solid;
    padding: 10px;
    border-radius: 7px;
    background: brown;
    color: white;
    margin: 20px;
    font-size: large;
    
}

@media screen and (max-width: 900px){
    .button{
        font-size: medium;
    }
}