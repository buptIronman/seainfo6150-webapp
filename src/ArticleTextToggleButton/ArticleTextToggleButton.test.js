import React from 'react';
import ArticleTextToggleButton from "./ArticleTextToggleButton";
import App from "../App";
import {fireEvent} from "@testing-library/dom";


describe("ArticleListItem test", () => {
    it("click button to show more", () => {
        const { container, getByText } = render(<App/>);
        const button = getByText("show more")
        fireEvent.click(button);
        expect(container).toMatchSnapshot();

    });

})