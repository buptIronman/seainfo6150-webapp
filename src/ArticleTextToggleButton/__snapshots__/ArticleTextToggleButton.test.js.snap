// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem test click button to show more 1`] = `
<div>
  <div>
    <button
      class="button"
    >
      show less
    </button>
  </div>
</div>
`;
